File: finalAssignmentBackend/src/main/java/com/tutict/finalassignmentbackend/config/statemachine/ProgressEvent.java
package com.tutict.finalassignmentbackend.config.statemachine;

// 事件枚举
public enum ProgressEvent {
    SUBMIT,      // 提交新进度
    START_PROCESSING, // 开始处理
    COMPLETE,    // 完成
    ARCHIVE,     // 归档
    REOPEN       // 重新打开（可选）
}

File: final_assignment_backend_go/project/configs/statemachine/progress_event.go
package statemachine

// ProgressEvent 定义状态机可触发的事件
type ProgressEvent string

const (
	EventSubmit          ProgressEvent = "SUBMIT"
	EventStartProcessing ProgressEvent = "START_PROCESSING"
	EventComplete        ProgressEvent = "COMPLETE"
	EventArchive         ProgressEvent = "ARCHIVE"
	EventReopen          ProgressEvent = "REOPEN"
)
